compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUSplitReduction.cpp

```diff
@@ -24,6 +24,129 @@ namespace mlir {
 namespace iree_compiler {
 namespace {
 
+/// Make sure that
+/// - the pass has not been applied before
+/// - has tensor semantics
+/// - number of reduction loops == 1
+/// - has exactly 1 output
+/// - index map has only projected permutations
+/// - is a linalg generic op
+/// - has exactly 1 input
+/// - if enableReductionReordering is not set, then operand is an int
+/// - innermost dimension of the input operand is reduction
+/// TODO: support named ops, numInputs > 1, and modify lastDim check below
+/// accordingly. If fpReductionReordering is not enabled by default, it must
+/// be an integer or index type to proceed to allow associative reordering.
+LogicalResult splitReductionPrecondition(Operation *op,
```

Comment:
Maybe not really great, but I think its ok to pass a `RewriterBase` as an operand here just to be able to replace `LLVM_DEBUG` with `rewriter.notifyMatchFailure()`.